### PR TITLE
doc: document submitting changes to modules

### PR DESCRIPTION
**Pull request by nashif**
Instructions for submitting changes to modules and now they will be 
integrated with the main tree.

### CODE REVIEW

File: doc/application/index.rst
Comment (mbolivar) [inline doc/application/index.rst]: Delete "below" since it's no longer below.

File: doc/guides/index.rst
Comment (mbolivar) [inline doc/guides/index.rst]: It looks like these are in alphabetical order by document title (not file name); putting this at the end breaks that.

File: doc/guides/modules.rst
Comment (mbolivar) [inline doc/guides/modules.rst]: These topologies etc belong in the west document, not here; zephyr modules are a build system source code organization artifact which is orthogonal to multirepo. Not all repositories in the west manifest are zephyr modules (like net-tools).

File: doc/guides/modules.rst
Comment (mbolivar) [inline doc/guides/modules.rst]: Maybe "will be created by the project team"?

s/initiated/initialized/

File: doc/guides/modules.rst
Comment (mbolivar) [inline doc/guides/modules.rst]: in the zephyr repository issue tracker (a link would be good too)

File: doc/guides/modules.rst
Comment (mbolivar) [inline doc/guides/modules.rst]: nitpick: GitHub

File: doc/guides/modules.rst
Comment (mbolivar) [inline doc/guides/modules.rst]: s/Code/code, etc./ (note lowercase)

File: doc/guides/modules.rst
Comment (mbolivar) [inline doc/guides/modules.rst]: s/Submit a new entry into the/Add the project to/

File: doc/guides/modules.rst
Comment (mbolivar) [inline doc/guides/modules.rst]: They need to update this before merge with the final revision, right?

File: doc/guides/modules.rst
Comment (mbolivar) [inline doc/guides/modules.rst]: I think it's worth noting here that requesting the pull head is for CI to be able to test it out. Then mention we'll merge the module code PR and then they need to update the west.yml PR with the final revision before we can merge that, making the module an official part of the project.

File: doc/guides/modules.rst
Comment (mbolivar) [inline doc/guides/modules.rst]: pull request

File: doc/guides/modules.rst
Comment (mbolivar) [inline doc/guides/modules.rst]: pull request

File: doc/guides/modules.rst
Comment (mbolivar) [inline doc/guides/modules.rst]: Same deal here, the additional post-merge update steps to the revision need documenting too.

File: doc/guides/modules.rst
Comment (mbolivar) [inline doc/guides/modules.rst]: submitter

File: doc/guides/modules.rst
Comment (mbolivar) [inline doc/guides/modules.rst]: Ah, I see you have it here; that's too late IMO. I think at least adding steps above with a `:ref:` to this section would make it more clear that it's part of the process for both adding new and modifying existing modules.

File: doc/guides/west/repo-tool.rst
Comment (mbolivar) [inline doc/guides/west/repo-tool.rst]: Why did you remove this from the west documentation then put it back?

Can you clean this up in the commits?

**Review by mbolivar — CHANGES_REQUESTED** [review]

File: doc/guides/modules.rst
Comment (aescolar) [reply doc/guides/modules.rst]: +the + submitter or by -the-+a+ maintainer

File: doc/guides/modules.rst
Comment (aescolar) [inline doc/guides/modules.rst]: > commit hash 

Question: Is the intention that the zephyr manifest will always point to hashes and never to tags of modules?

**Review by aescolar — DISMISSED** [review]
A nit, a question & a typo on the commit msg. "now" => "how"

**Comment by aescolar** [comment]
@nashif : Question: Is the plan then that CI will be triggered from PRs in the main repo?

File: doc/guides/modules.rst
Comment (nashif) [reply doc/guides/modules.rst]: both will work, a tag is a commit after all. we do not create tags for each merged PR in the modules now.

**Comment by nashif** [comment]
> @nashif : Question: Is the plan then that CI will be triggered from PRs in the main repo?

yes, CI on modules will be added just for compliance, but not for building and integration with zephyr.

**Comment by aescolar** [comment]
@nashif Isnt the commit description meant to be:
"Instructions for submitting changes to modules and -n-+h+ow they will be
integrated with the main tree."

**Comment by nashif** [comment]
> @nashif Isnt the commit description meant to be:
> "Instructions for submitting changes to modules and -n-+h+ow they will be
> integrated with the main tree."

fixed

**Review by carlescufi — CHANGES_REQUESTED** [review]
Nice job, few comments added

File: doc/guides/modules.rst
Comment (mbolivar) [reply doc/guides/modules.rst]: I think it would be nice to start tagging the released versions of each module (i.e. tag the version of each module used for Zephyr v1.15.0, etc.) so it would be easy to see using plain `git log` in each module repository what to use. But that's for the next release manager, not regular contributors (@ioannisg, thoughts?)

File: doc/contribute/index.rst
Comment (dbkinder) [inline doc/contribute/index.rst]:
```suggestion
Follow the guidelines in the :ref:`modules` section for contributing
```

File: doc/guides/modules.rst
Comment (dbkinder) [inline doc/guides/modules.rst]:
```suggestion
Zephyr build system, as described in more detail in other sections on
```

File: doc/guides/modules.rst
Comment (dbkinder) [inline doc/guides/modules.rst]:
```suggestion
Zephyr depends on several categories of modules, including:
```

File: doc/guides/modules.rst
Comment (dbkinder) [inline doc/guides/modules.rst]:
```suggestion
- Otherwise (i.e. if the project has no :file:`zephyr/module.yml`), the
```

File: doc/guides/modules.rst
Comment (dbkinder) [inline doc/guides/modules.rst]: lower case Z since this is a repo name, right?

File: doc/guides/modules.rst
Comment (dbkinder) [inline doc/guides/modules.rst]:
```suggestion
of the tree we use the commit hash, a tag, or a branch name. For pull requests
```

File: doc/guides/modules.rst
Comment (dbkinder) [reply doc/guides/modules.rst]: Actually, case of headings isn't consistent in this document.  I'd recommend just using a leading capital letter and then lower-case words throughout.

File: doc/guides/modules.rst
Comment (dbkinder) [reply doc/guides/modules.rst]: also, uppercase P on propose

File: doc/guides/modules.rst
Comment (dbkinder) [inline doc/guides/modules.rst]:
```suggestion
Where 23 in the example above indicated the pull request number submitted to the
```

File: doc/guides/modules.rst
Comment (dbkinder) [inline doc/guides/modules.rst]:
```suggestion
Where 23 in the example above indicated the pull request number submitted to the
```

**Review by dbkinder — CHANGES_REQUESTED** [review]

File: doc/guides/modules.rst
Comment (mbolivar) [inline doc/guides/modules.rst]: s/initiated/initialized/

**Review by mbolivar — COMMENTED** [review]
@nashif I've marked resolved everything I see that was addressed in my original comments; the other stuff is still open AFAICT

File: doc/guides/modules.rst
Comment (erwango) [inline doc/guides/modules.rst]: Shouldn't it be something like `The manifest should not point to repositories `**not**` maintained under the Zephyr project` ? Else I don't understand how this could match with previous sentence.

File: doc/guides/modules.rst
Comment (erwango) [inline doc/guides/modules.rst]: Will the modules be listed inside CODEOWNERS? If not, how will ownership be treated ?

File: doc/guides/modules.rst
Comment (carlescufi) [inline doc/guides/modules.rst]: I would change this to `Module Inclusion`, since we are not really initializing here, but rather including them in the build

File: doc/guides/modules.rst
Comment (carlescufi) [inline doc/guides/modules.rst]: Technically I think those are called "users" in GitHub terminology, not "projects". Also I would mention the username explicitly "zephyrproject-rtos"

**Review by carlescufi — CHANGES_REQUESTED** [review]

**Review by erwango — CHANGES_REQUESTED** [review]

File: doc/guides/modules.rst
Comment (nashif) [reply doc/guides/modules.rst]: organisation actually.

File: doc/guides/modules.rst
Comment (nashif) [reply doc/guides/modules.rst]: CODEOWNERS is per repo, for modules I do not think this will be needed, Github will figure this on its own. Each repo will have collaborators and members that will be responsible for reviews and changes.

File: doc/guides/modules.rst
Comment (carlescufi) [reply doc/guides/modules.rst]: Indeed

**Review by carlescufi — APPROVED** [review]

File: doc/guides/modules.rst
Comment (erwango) [reply doc/guides/modules.rst]: ok, txs for feedback!

**Review by marc-hb — COMMENTED** [review]
For anyone who didn't know this (I didn't until very recently, thx @carlescufi), the documentation build of this PR is available here:
https://builds.zephyrproject.org/zephyrproject-rtos/zephyr/16081/docs/application/index.html#important-build-system-variables
So much easier to review.

There's another thing that would make the review much easier: moving the content without changing it in a first commit; then making changes in different commit(s). This would make changes visible.

File: doc/guides/modules.rst
Comment (dbkinder) [inline doc/guides/modules.rst]: delete this line (extra "including")

**Review by dbkinder — CHANGES_REQUESTED** [review]
also looks like some of @mbolivar comments weren't addressed

**Comment by nashif** [comment]
> also looks like some of @mbolivar comments weren't addressed

github was hiding them, now they are.

**Pull request merged**
